### PR TITLE
Goa 2446 add name aspect to ontology slims

### PR DESCRIPTION
**Pull request by eddturner**

### CODE REVIEW

File: quickgo-client/src/main/java/uk/ac/ebi/quickgo/client/model/presets/PropertiesItem.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * A DTO that allows for the definition of dynamic properties.
+ *
+ * Created 05/12/16
+ * @author Edd
+ */
+public class PropertiesItem {
+    private final String id;
```
Comment (rantunesboreas) [inline quickgo-client/src/main/java/uk/ac/ebi/quickgo/client/model/presets/PropertiesItem.java]: What is the id for?

File: quickgo-client/src/main/java/uk/ac/ebi/quickgo/client/model/presets/impl/CompositePresetImpl.java [hunk 2]
```diff
@@ -127,31 +128,47 @@ private PresetItem transformGroupedEntryToPresetItem(PresetType presetType,
...
-
-        ifPresetItemMatchesThenApply(groupedEntry.getValue(),
-                p -> p.getDescription() != null && !p.getDescription().trim().isEmpty(),
-                p -> presetBuilder.withDescription(p.getDescription()));
-
-        ifPresetItemMatchesThenApply(groupedEntry.getValue(),
-                p -> p.getUrl() != null && !p.getUrl().trim().isEmpty(),
-                p -> presetBuilder.withUrl(p.getUrl()));
-
         if (presetType == GO_SLIMS_SETS) {
```
Comment (rantunesboreas) [inline quickgo-client/src/main/java/uk/ac/ebi/quickgo/client/model/presets/impl/CompositePresetImpl.java]: I don't think we should have this if statement, it is a specialization that shouldn't be done here. Why not create a curry function and have the specific behaviour added at the getGoSlimSets method?

File: quickgo-client/src/main/java/uk/ac/ebi/quickgo/client/model/presets/PresetItem.java [hunk 3]
```diff
@@ -22,15 +19,15 @@
     private Integer relevancy;
     private String id;
     private String url;
-    private List<String> associations;
+    private List<PropertiesItem> associations;
```
Comment (rantunesboreas) [inline quickgo-client/src/main/java/uk/ac/ebi/quickgo/client/model/presets/PresetItem.java]: Why have the associations as PresetItems and create a dynamicProps state variable?

File: quickgo-client/src/main/java/uk/ac/ebi/quickgo/client/model/presets/impl/CompositePresetImpl.java [hunk 4]
```diff
@@ -183,21 +201,46 @@ private void ifPresetItemMatchesThenApply(
         ASPECTS
     }
 
+    public enum SlimAdditionalProperty {
```
Comment (rantunesboreas) [inline quickgo-client/src/main/java/uk/ac/ebi/quickgo/client/model/presets/impl/CompositePresetImpl.java]: Maybe we should moves this classes out of the CompositePresetImpl, and leave them Package private.

File: quickgo-client/src/main/java/uk/ac/ebi/quickgo/client/service/loader/presets/slimsets/GOSlimSetPresetsConfig.java [hunk 5]
```diff
@@ -72,23 +69,20 @@ public Step goSlimSetStep(
...
      */
     private ItemWriter<RawNamedPreset> rawPresetWriter(CompositePresetImpl presets) {
-        return rawItemList -> rawItemList.forEach(rawItem -> {
-            presets.addPreset(CompositePresetImpl.PresetType.GO_SLIMS_SETS,
-                    PresetItem.createWithName(rawItem.name)
-                            .withId(rawItem.id)
-                            .build());
-        });
+        return rawItemList -> rawItemList.forEach(rawItem ->
+                presets.addPreset(CompositePresetImpl.PresetType.GO_SLIMS_SETS,
```
Comment (rantunesboreas) [inline quickgo-client/src/main/java/uk/ac/ebi/quickgo/client/service/loader/presets/slimsets/GOSlimSetPresetsConfig.java]: This chaining of calls leads me to believe that PresetType should probably be extracted from CompositePresetImpl

File: quickgo-client/src/test/java/uk/ac/ebi/quickgo/client/model/presets/PropertiesItemTest.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        String id = null;
+        PropertiesItem.createWithId(id).build();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void emptyIdCausesException() {
+        String id = "";
+        PropertiesItem.createWithId(id).build();
+    }
+
```
Comment (rantunesboreas) [inline quickgo-client/src/test/java/uk/ac/ebi/quickgo/client/model/presets/PropertiesItemTest.java]: You should have a test that verifies that a valid id does not throw an IllegalArgumentException

File: quickgo-client/src/main/java/uk/ac/ebi/quickgo/client/model/presets/impl/CompositePresetImpl.java (hunk 4 above)
Comment (eddturner) [inline quickgo-client/src/main/java/uk/ac/ebi/quickgo/client/model/presets/impl/CompositePresetImpl.java]: done

File: quickgo-client/src/main/java/uk/ac/ebi/quickgo/client/model/presets/impl/CompositePresetImpl.java (hunk 2 above)
Comment (eddturner) [inline quickgo-client/src/main/java/uk/ac/ebi/quickgo/client/model/presets/impl/CompositePresetImpl.java]: done

File: quickgo-client/src/main/java/uk/ac/ebi/quickgo/client/model/presets/PresetItem.java (hunk 3 above)
Comment (eddturner) [inline quickgo-client/src/main/java/uk/ac/ebi/quickgo/client/model/presets/PresetItem.java]: no longer applicable after refactoring PropertiesItem -> PresetItem

**Pull request merged**